scripts/check_cpp_format.sh

```diff
@@ -1,41 +1,12 @@
 #!/bin/bash
 
-CLANG_FORMAT=${CLANG_FORMAT:-clang-format}
-set -e
-
-if [ $# -eq 0 ]; then
-  echo "$(basename $0) [-fix] <paths>" >&2
-  exit 1
-fi
-
-FIX=
-if [ "$1" == "-fix" ]; then
-  FIX=1
-  shift 1
-fi
-
-ROOTS="$@"
-FAILED=
-PRUNE_PATHS=${PRUNE_PATHS:-}
-PRUNE_NAMES="build* .#*"
-
-emit_prunes() {
-  { for p in ${PRUNE_PATHS}; do echo "-path ${p} -prune -o"; done; \
-    for p in ${PRUNE_NAMES}; do echo "-name ${p} -prune -o"; done; } | xargs
-}
-
-while read -d '' filename; do
-  if [ -n "${FIX}" ]; then
-    echo "fixing ${filename}"
-    ${CLANG_FORMAT} -style=file -i "${filename}"
-  else
-    if ${CLANG_FORMAT} -style=file -output-replacements-xml "${filename}" | grep '<replacement ' > /dev/null; then
-      echo "${filename} NOT OK"
-      FAILED=1
-    fi
-  fi
-done < <(find ${ROOTS} $(emit_prunes) -name '*.cpp' -print0 -o -name '*.h' -print0 -o -name '*.cu' -print0 -o -name '*.cuh' -print0)
-
-if [ -n "${FAILED}" ]; then
-  exit 1
-fi
+# Taken from https://stackoverflow.com/a/246128/1915854
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  DIR="$( cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1 && pwd )"
+  SOURCE="$(readlink "$SOURCE")"
+  [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+DIR="$( cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1 && pwd )"
+
+python3 ${DIR}/check_cpp_format.py "$@"
```

Comment:
I don't think we need this wrapper. People can call `check_cpp_format.py` instead.

Reply:
Without a major reorganization of the other scripts in the codebase, we need this wrapper. The problem is that `scripts/check_all.sh` is a symlink pointing to `external/katana/scripts/check_all.sh`, so a user can call it from both directories and we need `check_cpp_format.sh` in both directories. An alternative to providing this wrapper would be to place another `check_cpp_format.py` script in the enterprise repo, that would just call the Python script from the open repo. Then we would be able to get rid of `check_cpp_format.sh` in both directories.

Reply:
Couldn't you just put a symlink in the enterprise directory just like there are symlinks to other scripts?

Reply:
I would have to put a symlink to the specific Python script as well as the symlink to the directory containing the abstract package, right? If you think it's a good idea, I can do it in https://github.com/KatanaGraph/katana/pull/432 as I found some incompatibilities with Python 3.6 anyway.

Reply:
I think it's reasonable, yes. But it's up to you. Not urgent.

Reply:
Ok, I'll create symlinks in my next open PR.